packages/connectivity/src/http-agent/http-agent.spec.ts

```diff
@@ -258,4 +259,63 @@ describe('getAgentConfig', () => {
 
     expect(() => getAgentConfig(destination)).toThrow();
   });
+
+  describe('mTLS', () => {
+    describe('on CloudFoundry', () => {
+      beforeAll(() => {
+        mock({
+          'cf-crypto': {
+            'cf-cert': 'my-cert',
+            'cf-key': 'my-key'
+          }
+        });
+
+        process.env.CF_INSTANCE_CERT = 'cf-crypto/cf-cert';
+        process.env.CF_INSTANCE_KEY = 'cf-crypto/cf-key';
+      });
+
+      afterAll(() => {
+        mock.restore();
+
+        delete process.env.CF_INSTANCE_CERT;
+        delete process.env.CF_INSTANCE_KEY;
+      });
+
+      it('returns an object with key "httpsAgent" and mTLS options for destinations with inferMtls option and environment variables are present', async () => {
+        const destination: HttpDestination = {
+          url: 'https://example.com',
+          mtls: true
+        };
+        const actual = getAgentConfig(destination)['httpsAgent'].options;
+
+        expect(actual.cert).toEqual('my-cert');
+        expect(actual.key).toEqual('my-key');
+        expect(actual.pfx).not.toBeDefined();
+        expect(actual.passphrase).not.toBeDefined();
+      });
+    });
+
+    it('returns an object with key "httpsAgent" and mTLS options for destinations with inferMtls option and environment variables are not present', async () => {
+      const destination: HttpDestination = {
```

Comment:
[q] If `inferMtls` option is true but `CF_INSTANCE_CERT` and `CF_INSTANCE_KEY` are not present in the env variables, should this not lead to any warning or error?

Reply:
I don't have a strong opinion here, my thinking was that 'infer' does not guarantee it will do something so I'm not sure if we should really warn if it does not happen.

Reply:
i'm also in favor of throwing a warning in this case

Reply:
added a log and adapted the unit test to assert it is there